service/deploy.py

```diff
@@ -198,15 +196,49 @@ def configure_ansible(logger):
 
 def build_host_name(ip):
     """
+    FIXME: return FQDN (on flag?)
     Return the host name
     * iPlant or raw
     """
-    return iplant_hostname(ip)
+    if not hasattr(settings, 'DEFAULT_IP_LOOKUP'):
+        return iplant_hostname(ip)
+    hostname_fn_str = settings.DEFAULT_IP_LOOKUP
+    hostname_fn = settings._get_method_for_string(hostname_fn_str, the_globals=globals())
+    return hostname_fn(ip)
+
+
+def split_ip_address(ip):
+    try:
+        regex = re.compile(
+            "(?P<one>[0-9]+)\.(?P<two>[0-9]+)\."
+            "(?P<three>[0-9]+)\.(?P<four>[0-9]+)")
+        r = regex.search(ip)
+        (one, two, three, four) = r.groups()
+        return (one, two, three, four)
+    except Exception:
+        raise Exception("IP <%s> is not of the format VVV.XXX.YYY.ZZZ" % ip)
+
+
+def jetstream_hostname(ip):
+    """
+    INPUT: IP Address (str in form: VVV.XXX.YYY.ZZZ)
+    js-YYY-ZZZ.jetstream-cloud.org
+    """
+    prefix = "js-"
+    separator = "-"
+    list_of_subnet = split_ip_address(ip)
+    return "%s%s%s%s.%s" % (
+        prefix, list_of_subnet[2], separator, list_of_subnet[3],
+        "jetstream-cloud.org")
```

Comment:
Is there any way we might be able to make this a configuration-driven strategy?

I'm thinking of how DRF does things like paging arguments or something along those lines...